### PR TITLE
Control plane mtu

### PR DESCRIPTION
**Pull request by fcrisciani**
**- What I did**
Add the control plane MTU option in the daemon config. This will be useful in clusters with
MTU higher than 1500. The first user of this option will be libnetwork that will use it to
seed the networkDB library.

**- Description for the changelog**

Add the control plane MTU option in the daemon config

**- A picture of a cute animal (not mandatory but encouraged)**

funny-cute-animal-pics-part119-17-w630

### CODE REVIEW

**Comment by fcrisciani** [comment]
Waiting for the vendoring of the libnetwork side, but will take CR or comments in the mean time

**Comment by friism** [comment]
@fcrisciani will this help with this problem? https://github.com/moby/moby/issues/33596

**Comment by fcrisciani** [comment]
@friism unfortunately nope, this is for control plane traffic, that is instead data plane traffic issue. Good to know though because gossip cap the packet at 1400, so this feature with Microsoft node most likely won't work till they fix it.

**Comment by fcrisciani** [comment]
@friism looks like we support MTU on the overlay: https://github.com/docker/libnetwork/pull/1349, did they try this?

File: daemon/config/config.go [hunk 1]
```diff
@@ -168,6 +168,9 @@ type CommonConfig struct {
 	ValuesSet map[string]interface{}
 
 	Experimental bool `json:"experimental"` // Experimental indicates whether experimental features should be exposed or not
+
+	// ControlPlaneMTU allows to specify the control plane MTU, this will allow to optimize the network use in some components
+	ControlPlaneMTU int `json:"control-plane-mtu,omitempty"`
```
Comment (cpuguy83) [inline daemon/config/config.go]: `NetworkControlPlaneMTU`?

File: cmd/dockerd/config.go [hunk 2]
```diff
@@ -62,6 +62,8 @@ func installCommonConfigFlags(conf *config.Config, flags *pflag.FlagSet) {
 
 	flags.StringVar(&conf.MetricsAddress, "metrics-addr", "", "Set default address and port to serve the metrics api on")
 
+	flags.IntVar(&conf.ControlPlaneMTU, "control-plane-mtu", config.DefaultNetworkMtu, "Control plane MTU")
```
Comment (cpuguy83) [inline cmd/dockerd/config.go]: `network-control-plane-mtu`?

File: daemon/config/config.go (hunk 1 above)
Comment (fcrisciani) [inline daemon/config/config.go]: sure, I will update it with once the libnetwork side is ready

File: cmd/dockerd/config.go (hunk 2 above)
Comment (fcrisciani) [inline cmd/dockerd/config.go]: ok

**Comment by fcrisciani** [comment]
@cpuguy83 ready for a second pass

**Review by cpuguy83 — APPROVED** [review]
Code and Design LGTM

**Comment by vieux** [comment]
LGTM

**Comment by thaJeztah** [comment]
> If the value is not specified or is lower than 1500 the logic will set it to the default.

@fcrisciani should values lower than 1500 produce an error instead of silently ignoring? I can see it being unexpected if someone sets a lower value and it's not being used (or; if lower values become valid in future, they become active after updating to a newer version)

**Comment by fcrisciani** [comment]
@thaJeztah in reality that is a comment that did not get updated, on the libnetwork side we decided to allow any configuration but we will log a warning (https://github.com/docker/libnetwork/blob/bump_17.07/config/config.go#L231)

```
if exp < 1500 {
     logrus.Warnf("Received a MTU of %d, this value is very low, the network control plane can misbehave", exp)
}
```

**Pull request merged**